proc/sort.go

```diff
@@ -93,85 +227,194 @@ func guessSortField(rec *zng.Record) string {
 	return "ts"
 }
 
-func (s *Sort) Pull() (zbuf.Batch, error) {
+// runManager manages runs (files of sorted records).
+type runManager struct {
+	runs       []*runFile
+	runIndices map[*runFile]int
+	sortFn     expr.SortFn
+	tempDir    string
+	zctx       *resolver.Context
+}
+
+// newRunManager creates a temporary directory.  Call cleanup to remove it.
+func newRunManager(sortFn expr.SortFn) (*runManager, error) {
```

Comment:
"Neither do people light a lamp and put it under a bowl. Instead they put it on its stand, and it gives light to everyone in the house." Why hide this function? It would be cool to have this return a `zbuf.Reader`- perhaps this function could even live in zbuf. I imagine the signature could some like:
```golang
func FileMergeSort(ctx context.Context, sortFn expr.SortFn, r zbuf.Reader, maxSize int64) (zbuf.Reader, error)
```

The function would always keep the last record set in memory so if the amount of total records is less then maxSize, no records would ever hit disk.

Reply:
This is a good idea, and I'm going to do it in another PR.